src/main/java/sirius/pasta/tagliatelle/emitter/LoopEmitter.java

```diff
@@ -63,8 +67,10 @@ public Emitter reduce() {
      * @param compilationContext the context used to report errors to
      */
     public void verify(CompilationContext compilationContext) {
-        if (!Iterable.class.isAssignableFrom(iterableExpression.getType())) {
-            compilationContext.error(startOfBlock, "A for loop must have an Iterable as expression");
+        if (!Iterable.class.isAssignableFrom(iterableExpression.getType()) && iterableExpression.getGenericType()
```

Comment:
```suggestion
        if (!Iterable.class.isAssignableFrom(iterableExpression.getType()) && !iterableExpression.getGenericType()
```

Oder ist noch zu früh am Morgen?